### PR TITLE
fix: don't setup-golang when downloading a binary

### LINKED ISSUE #3073: [feature] avoid setup-golang tools when downloading a binary
**Is your feature request related to a problem? Please describe.**

I'm currently getting warnings from setup-golang action, due to having already setup golang with a different version and cache location. 

That slsa generator needs it when its downloading a binary seems erroneous and is a causing this spurious message as well as wasting ci time.

https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/actions/generate-builder/action.yml

instead the setup-golang step should be conditional on the input if the tool is to be built.

Screenshot 2024-01-10 at 6 12 44 AM

### PR DESCRIPTION
**Pull request by kapilt**
closes #3073

### CODE REVIEW

File: .github/actions/generate-builder/action.yml [hunk 1]
```diff
@@ -77,6 +77,7 @@ runs:
 
     - name: Set up Go environment
       uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+      if: $${{ inputs.compile-builder }}
```
[the remark this answers is not in the document]
Comment (kapilt) [reply .github/actions/generate-builder/action.yml:80]: afaics it was being passed as a boolean re required arg vs string.

[the remark this answers is not in the document]
Comment (ianlewis) [reply .github/actions/generate-builder/action.yml:80]: All inputs in GitHub Actions are basically strings so I think we'll need to do `compile-builder == 'true'` otherwise it's a test whether the string is empty or not.

https://stackoverflow.com/questions/76292948/github-action-boolean-input-with-default-value
